samcli/lib/intrinsic_resolver/intrinsics_symbol_table.py

```diff
@@ -0,0 +1,402 @@
+        account_id = self.logical_id_translator.get(
+            IntrinsicsSymbolTable.AWS_ACCOUNT_ID) or self.handle_pseudo_account_id()
+        partition_name = self.handle_pseudo_partition()
+        function_name = logical_id
+        function_name = self.logical_id_translator.get(function_name) or function_name
+        str_format = "arn:{partition_name}:{service_name}:{aws_region}:{account_id}:{function_name}"
+        if service_name == "lambda":
+            str_format = "arn:{partition_name}:{service_name}:{aws_region}:{account_id}:function:{function_name}"
+
+        return str_format.format(
+            partition_name=partition_name,
+            service_name=service_name,
+            aws_region=aws_region,
+            account_id=account_id,
+            function_name=function_name)
+
+    def get_translation(self, logical_id, resource_attributes=IntrinsicResolver.REF):
+        """
+        This gets the logical_id_translation of the logical id and resource_attributes.
+
+        Parameters
+        ----------
+        logical_id: str
+            This is the logical id of the resource in question
+        resource_attributes: str
+            This is the attribute required. By default, it is a REF type
+
+        Returns
+        --------
+        This returns the translated item if it already exists
+
+        """
+        logical_id_item = self.logical_id_translator.get(logical_id, {})
+        if isinstance(logical_id_item, string_types):
+            if resource_attributes != IntrinsicResolver.REF and resource_attributes != "":
+                return None
+            return logical_id_item
+        return logical_id_item.get(resource_attributes)
+
+    @staticmethod
+    def get_availability_zone(region):
+        """
+        This gets the availability zone from the the specified region
+
+        Parameters
+        -----------
+        region: str
+            The specified region from the SymbolTable region
+
+        Return
+        -------
+        The list of availability zones for the specified region
+        """
+        return IntrinsicsSymbolTable.REGIONS.get(region)
+
+    def handle_pseudo_account_id(self):
+        """
+        This gets a default account id from SamBaseProvider.
+        Return
+        -------
+        A pseudo account id
+        """
+        return SamBaseProvider.DEFAULT_PSEUDO_PARAM_VALUES.get(IntrinsicsSymbolTable.AWS_ACCOUNT_ID)
+
+    def handle_pseudo_region(self):
+        """
+        Gets the region from the environment and defaults to a the default region from the global variables.
+
+        This is only run if it is not specified by the logical_id_translator as a default.
+
+        Return
+        -------
+        The region from the environment or a default one
+        """
+        return os.getenv("AWS_REGION") or SamBaseProvider.DEFAULT_PSEUDO_PARAM_VALUES.get(
```

Comment:
This only accounts for if they set `AWS_REGION`, what happens if a customer passes `--region` into the command. How do we resolve to that region instead of the default or one set in an env var?

Reply:
My plan for this during integration would be to add to add it to the logical_id_translator and override it.